### PR TITLE
Prometheus based metrics and monitoring for KFServing models

### PR DESCRIPTION
**Pull request by sriumcp**
**What this PR does / why we need it**:
This PR provides samples and instructions for getting started with Prometheus based metrics and monitoring for KFServing models. While the initial focus of this PR is on metrics auto-generated by KNative's queue-proxy container, we/the community can expand on this documentation in the future with documentation on how to collect and access ML/business metrics as well.

This is a pure documentation PR. 

**Special notes for your reviewer**:

@animeshsingh 

**Release note**:

```release-note
Added documentation for Prometheus based metrics and monitoring for KFServing models. 
Link: https://github.com/kubeflow/kfserving/tree/master/docs/samples/metrics-and-monitoring
```

### CODE REVIEW

File: docs/samples/metrics-and-monitoring/prometheus/clusterrolebinding.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: prometheus
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: prometheus
+subjects:
+- kind: ServiceAccount
+  name: prometheus
+  namespace: default
```
Comment (Tomcli) [inline docs/samples/metrics-and-monitoring/prometheus/clusterrolebinding.yaml:12]: Do you mean `kfserving-monitoring` over here? Maybe we should parameterize it so users can also use it on the istio-system prometheus.

File: docs/samples/metrics-and-monitoring/README.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+```
+while clear; do \
+  curl -v \
+  -H "Host: ${SERVICE_HOSTNAME}" \
+  -d @./iris-input.json \
+  http://${INGRESS_HOST}:${INGRESS_PORT}/v2/models/sklearn-iris/infer
+  sleep 0.3
+done
+```
+6. In a separate terminal, port forward the Prometheus service.
```
Comment (animeshsingh) [inline docs/samples/metrics-and-monitoring/README.md:42]: We should probably mention port-forwarding is optional, if ingress ip is accessible

File: docs/samples/metrics-and-monitoring/README.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+```shell
+kubectl port-forward service/prometheus-operated -n kfserving-monitoring 9090:9090
+```
+7. Access Prometheus UI in your browser at http://localhost:9090
```
Comment (animeshsingh) [inline docs/samples/metrics-and-monitoring/README.md:46]:
```suggestion
7. Access Prometheus UI in your browser at http://localhost:9090 or http://${INGRESS_HOST}:9090
```

**Review by animeshsingh — CHANGES_REQUESTED** [review]

File: docs/samples/metrics-and-monitoring/prometheus/clusterrolebinding.yaml (hunk 1 above)
Comment (sriumcp) [reply docs/samples/metrics-and-monitoring/prometheus/clusterrolebinding.yaml]: The `namespace` field in that line is simply a place holder and its value will be overwritten by the top-level kustomization. Nevertheless, the `default` value might be confusing, so I replaced it with `kfserving-monitoring` in my latest commit.

I also added explicit instructions for how to kustomize the install namespace.

File: docs/samples/metrics-and-monitoring/README.md (hunk 2 above)
Comment (sriumcp) [reply docs/samples/metrics-and-monitoring/README.md:44]: 1. Port-forward will make Prometheus available locally, only within the machine where the port-forwarding command is issue.

2. Istio-ingress configuration for Prometheus would make it available to the entire world outside of the Kubernetes cluster.

Perhaps it is better to stick to port-forward for security considerations?

File: docs/samples/metrics-and-monitoring/README.md (hunk 3 above)
Comment (sriumcp) [reply docs/samples/metrics-and-monitoring/README.md:48]: Access Prometheus through Istio-ingress actually requires reconfiguring the Istio ingress gateway. By default, Istio-ingress is not set up to expose Prometheus service externally (please see comment above).

**Review by animeshsingh — APPROVED** [review]

**Pull request merged**